src/sagas/permissions.js

```diff
@@ -0,0 +1,54 @@
+import { Platform } from 'react-native';
+import { check, PERMISSIONS, request, RESULTS } from 'react-native-permissions';
+import {
+  all,
+  put,
+  select,
+  takeLatest,
+} from 'redux-saga/effects';
+import { setCameraAvailable, types } from '../actions';
+
+/**
+ * Checks the current permissions for the camera and sets the results on redux.
+ * If permission is not granted, but requireable, requests the permission from the user.
+ * Sets the redux property IS_CAMERA_AVAILABLE to True if the app has permission to use the camera
+ *
+ * For a complete description on all possible permition statuses,
+ * @see https://github.com/zoontek/react-native-permissions#permissions-statuses
+ */
+export function* requestCameraPermissions() {
+  const checkStatus = async (status) => {
+    switch (status) {
+      case RESULTS.GRANTED:
+        return true;
+      case RESULTS.DENIED: // The permission has not been requested / is denied but requestable
+        return checkStatus(await request(platformCameraPermission));
```

Comment:
❓ Can it fall in infinity loop? Have you tested the case in which the used denied the permission over and over again?

Reply:
Actually this implementation fixes an infinity loop that happened before it. The simplified permissions flow works like this:
- If the user has granted unlimited permission to the application, the status returned by `check()` is `GRANTED`
- In all other cases, the status is `DENIED`.

Upon `request()`, the result can then be either `GRANTED` or `BLOCKED`. The `DENIED` is never repeated.
So, there is no loop condition unless the `react-native-permissions` lib introduces a breaking change.

Reply:
Great!